### PR TITLE
docs: rename `dvc lock` -> `dvc freeze`

### PR DESCRIPTION
**Pull request by efiop**
Per https://github.com/iterative/dvc/pull/3915

> You may disregard these recommendations if you used the **Edit on GitHub** button from dvc.org to improve a doc in place.

❗ Please read the guidelines in the [Contributing to the Documentation](https://dvc.org/doc/user-guide/contributing/docs) list if you make any substantial changes to the documentation or JS engine.

🐛 Please make sure to mention `Fix #issue` (if applicable) in the description of the PR. This causes GitHub to close it automatically when the PR is merged.

Please choose to [allow us](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to edit your branch when creating the PR.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

### CODE REVIEW

**Comment by shcheklein** [comment]
things to take care of:

- [ ] add redirects lock -> freeze (redirects.json)
- [ ] rename in the prism dvc-commands.js
- [ ] fix completion scripts
- [ ] grep `dvc lock` - to make sure that we renamed it everywhere

**Comment by efiop** [comment]
@shcheklein Done.

**Comment by efiop** [comment]
Tests fail because they try to check links that are not yet deployed.

**Comment by shcheklein** [comment]
@efiop yep, it's a well know and unfortunate limitation, gonna fix this in #1000 /cc @casperdcl

File: content/docs/command-reference/freeze.md [hunk 1]
```diff
@@ -1,32 +1,32 @@
...
 
 ## Description
 
-`dvc lock` causes any DVC-file to be considered _not changed_ by `dvc status`
-and `dvc repro`. Stage reproduction will not execute regenerate
-<abbr>outputs</abbr> of locked stages, even if some dependencies have changed,
+`dvc freeze` causes any stage to be considered _not changed_ by `dvc status` and
+`dvc repro`. Stage reproduction will not execute the command or regenerate
+<abbr>outputs</abbr> of frizen stages, even if some dependencies have changed,
 and even if `--force` is provided.
```
Comment (jorgeorpinel) [inline content/docs/command-reference/freeze.md:20]: Thanks for fix! I think here we meant to simplify to just
```suggestion
`dvc repro`. Stage reproduction will not regenerate <abbr>outputs</abbr>
of frozen stages, even if some dependencies have changed, and even if
`--force` is provided.
```
Also, typo fixed ^ 😬

File: content/docs/sidebar.json [hunk 2]
```diff
@@ -256,8 +256,8 @@
         "slug": "list"
       },
       {
-        "label": "lock",
-        "slug": "lock"
+        "label": "freeze",
+        "slug": "freeze"
       },
```
Comment (jorgeorpinel) [inline content/docs/sidebar.json:261]: Yep but this has to be moved up between fetch and gc now 🙂

**Comment by efiop** [comment]
> Couple minor issues. Is this released though? Otherwise no rush no merge this, right? Thanks

@jorgeorpinel It is merged. I think current approach is to just merge 1.0 changes into master, no?

**Comment by jorgeorpinel** [comment]
> current approach is to just merge 1.0 changes into master, no?

I'm not sure. I thought we were timing all of these meaning just letting them be approved to merge them all at once when 1.0 is released.

**Comment by efiop** [comment]
@jorgeorpinel We have a mix of all of them right now. E.g. plots is from 1.0. It makes sense to continue merging.

**Comment by jorgeorpinel** [comment]
Post moretm review/ questions @efiop:

- What about phrase "always locked" used in the `import(_url)` cmd refs?
- What about `locked` field in YAML files? Did it change to `frozen`? (Should it?) — mentioned in examples and DVC files & dirs guide
- `.dvc/lock`, `.dvc/tmp/lock`, and `.dvc/tmp/rwlock` stay the same?
- `dvc show --locked` ?

Thanks

**Comment by efiop** [comment]
@jorgeorpinel 

> What about phrase "always locked" used in the import(_url) cmd refs?

Oops, missed that one. https://github.com/iterative/dvc.org/pull/1404 Thank you! :pray: 

> What about locked field in YAML files? Did it change to frozen? (Should it?) — mentioned in examples and DVC files & dirs guide
I didn't touch that one because it will need to be removed/reworked for 1.0. I'll leave a comment in https://github.com/iterative/dvc.org/pull/1370/files , since it is already there.

> .dvc/lock, .dvc/tmp/lock, and .dvc/tmp/rwlock stay the same?
Correct.

> dvc show --locked ?
no such command in 1.0, `dvc dag` replaced it and didn't implement `locked` yet

**Comment by jorgeorpinel** [comment]
Awesome.

**Comment by efiop** [comment]
> .dvc/lock, .dvc/tmp/lock, and .dvc/tmp/rwlock stay the same?

@jorgeorpinel Just to clarify: `.dvc/lock` is no more, it has been moved to `.dvc/tmp/lock`. https://dvc.org/doc/user-guide/dvc-files-and-directories already reflects that.

**Pull request merged**